packages/@aws-cdk/aws-redshift/lib/parameter-group.ts

```diff
@@ -22,6 +35,15 @@ abstract class ClusterParameterGroupBase extends Resource implements IClusterPar
    * The name of the parameter group
    */
   public abstract readonly clusterParameterGroupName: string;
+
+  /**
+   * Indicates if the parameter group parameters can be updated.
+   */
+  protected abstract updatableParameters: boolean;
+
+  public addParameter(_name: string, _value: string): AddParameterResult {
+    return { parameterAddedResult: AddParameterResultStatus.I_FAILURE };
```

Comment:
I thought about leaving the method as `abstract` , but wanted to remain consistent with the [changes I made to the cluster construct](https://github.com/dontirun/aws-cdk/blob/f2d50155a32cca0e5ad4b77002e2a9928794aaa9/packages/%40aws-cdk/aws-redshift/lib/cluster.ts#L375)

For `addToParameterGroup` in the `Cluster` construct, this method should fail for all imported resources since we can't add parameters to them.  The L2 `Cluster` class is the only class that it wouldn't fail on. Any of the static import methods such as [fromClusterAttributes](https://github.com/dontirun/aws-cdk/blob/f2d50155a32cca0e5ad4b77002e2a9928794aaa9/packages/%40aws-cdk/aws-redshift/lib/cluster.ts#L389) should fail since they are imported resources. In the future there could *potentially*  be more static methods that create imported resources from `ClusterBase` which inherit this method but I don't think there would be another instantiated Cluster class that extends `ClusterBase`; so I thought the default should be the failure.

I recognize that the [static method/ imported](https://github.com/dontirun/aws-cdk/blob/f2d50155a32cca0e5ad4b77002e2a9928794aaa9/packages/%40aws-cdk/aws-redshift/lib/parameter-group.ts#L112) class on `ClusterParameterGroup` is implemented slightly differently, implementing `IClusterParameterGroup` directly instead of extending `ClusterParameterGroupBase` (side note, I don't know why this is the case). Because of this difference I know that that each imported ParameterGroup would need to implement this method individually. I decided to keep the failure as a default here just to remain consistent with the `Cluster` implementation.